add unmount target + use printf over echo + update PHONY list

TheDen opened the pull request:
PR to update the `Makefile`, changes include

* Adds `unmount` target which ejects the Alacritty volume via `hdiutil`
* Replaces instances of `echo` with `printf`, so now it's consistent 
* Adds missing `all`, `test`, and now `unmount` to `.PHONY`

chrisduerr reviewed (changes requested):
Why?

TheDen commented:
> Why?

Which part are you asking about?

chrisduerr commented:
Why are you sending this PR? What is the motivation behind it? There has been zero demand for it to my knowledge.

TheDen commented:
> Why are you sending this PR? What is the motivation behind it? here has been zero demand for it to my knowledge.

I want to, was building and installing a lot, and idc

TheDen commented:
Closing this. A reviewer asking snarky one word replies about motives and demands of a PR that simply does minor updates to a `Makefile`...yeah @chrisduerr totally not toxic behaviour...

nixpulvis commented:
Asking for the motivation for a change is completely reasonable and he even elaborated when asked. As it isn't clearly solving any open issues or known sticking points for people, it can be said to be a pet idea you had. No shame in that, but also no expectation of being merged.

Let's avoid name calling shall we?

TheDen commented:
> Asking for the motivation for a change is completely reasonable and he even elaborated when asked.

Agree, but for me it was the tone. idk maybe I expected better comms from this project so it was jarring

> Let's avoid name calling shall we?

Agreed, but calling out someone's behaviour isn't the same thing as name calling

Look I'm sure chrisduerr isn't a jerk, I get busy and tired too and slip. It might be good to add a [PR template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository) with these common questions, there's an issues template in this repo, and I found it useful.

nixpulvis commented:
OK, so my curiosity is piqued.

@TheDen:

- Why are you adding a `test` `.PHONY` target when there is no `test`?
- Why are you changing `@echo` to `@printf` when it requires more code?
- Why are you adding a way to `detach` when there is no corresponding `attach`?

As for your last comment. I'd just say that each project and community has their own norms and expectations. I would read through other closed pull requests and issues to get a sense of things personally. It shouldn't really need to be stated that motivating a change is a good idea, especially when the change is somewhat unjustified, as I've pointed out above.

This isn't about being a jerk or not. This is about making good thoughtful changes to the source.

TheDen commented:
Okay, yeah maybe let's start over

>Why are you adding a test .PHONY target when there is no test?

I use [checkmake](https://github.com/mrtazz/checkmake) and `test` is a required phony target for the `minphony` rule. It passes everything else. I can remove it of course.

>Why are you changing @echo to @printf when it requires more code?

`printf` is also used in the `help` target, so IMO we ought to just use one rather than both `echo` and `printf`. Since `printf` offers better portability, I think it's the right choice.

>Why are you adding a way to detach when there is no corresponding attach?

The `hdiutil create` command under the `dmg` targets creates and mounts the volume. So after running `make install`, it's useful to unmount it via the cli, since `Finder.app` can be buggy when unmounting (at least for me). Maybe the naming of the target and whatnot could be better?

Motivation-wise I was building for [this](https://github.com/alacritty/alacritty/issues/6747) playing around with different [values](https://github.com/alacritty/alacritty/pull/6712/files) and changing my mind after installing it, it got annoying using `Finder.app`,and I saw improvement whilst editing the `Makefile` and thought I'd share them. I mean these are small details, and I usually focus on details, so  I didn't really think much of it and thought the changes were obvious which was perhaps an err on my part.

nixpulvis commented:
Strange, I even tested with `make dmg`:

```
Created 'Alacritty.app' in 'target/release/osx'
Packing disk image...
..........................
created: /Users/nixpulvis/Developer/alacritty/target/release/osx/Alacritty.dmg
Packed 'Alacritty.app' in 'target/release/osx'
```

Maybe I don't know where it's mounted, but I don't see it. Sorry, I'm not as familiar with macOS as I once was.

nixpulvis commented:
P.S. funny, I just commented on the horizontal scrolling myself here: https://github.com/alacritty/alacritty/pull/6625#issuecomment-1452990057 😉

nixpulvis commented:
Oh, you're doing a `make install`, well yea... all that does is open the DMG. I feel like this change is over the top macOS developer handholding though. Frankly, I'd rather see the `install` target removed before this PR merged, since I prefer less code to more code, especially in the build systems.

TheDen commented:
Yeah I think I shouldn't have been using `make install`, also subsequent multiple `make installs` creates multiple mounts
<img width="153" alt="image" src="https://user-images.githubusercontent.com/856001/222642467-096361b5-110a-4755-808e-846727457538.png">

```bash
$ df /Volumes/Alacritty*
Filesystem                      512-blocks      Used  Available Capacity iused      ifree %iused  Mounted on
/dev/disk4s1                         16800     11040       5760    66%      22 4294967257    0%   /Volumes/Alacritty
/dev/disk5s1                         16800     11040       5760    66%      22 4294967257    0%   /Volumes/Alacritty 1
/dev/disk6s1                         16800     11040       5760    66%      22 4294967257    0%   /Volumes/Alacritty 2
```

I think they're distinct mounts, and generally an unergonomic flow MacOS users have gotten used to with mounts. Agree on dropping the `install` target

TheDen commented:
Removed the `install` targets, and `unmount` target. Also dropped `test` from `.PHONY` since the project doesn't have `makecheck` as a pass req

Running `git grep 'make install'` returns nothing, so AFAIK it's not really used, or at least referenced outside the `Makefile`

chrisduerr reviewed (changes requested):
So now this PR basically just removes features that were previously supported? There's really no benefits being provided here.
